src/core/file.cc

```diff
@@ -206,7 +206,14 @@ posix_file_impl::flush() noexcept {
 
 future<struct stat>
 posix_file_impl::stat() noexcept {
-    return engine().fstat(_fd);
+    return engine()._thread_pool->submit<syscall_result_extra<struct stat>>([fd = _fd] {
+        struct stat st;
+        auto ret = ::fstat(fd, &st);
```

Comment:
I expect we'll need to move it back, as reactor::open_and_stat(), once we do this via io_uring. We need an abstraction in reactor once it can be done in more than one way.